msprime/mutations.py

```diff
@@ -83,14 +83,95 @@ class JukesCantor(MutationModel):
     """
 
     def __init__(self):
-        alleles = [b"A", b"C", b"T", b"G"]
+        alleles = [b"A", b"C", b"G", b"T"]
         root_distribution = [0.25, 0.25, 0.25, 0.25]
         transition_matrix = np.zeros((4, 4))
         transition_matrix[:] = 1 / 3
         np.fill_diagonal(transition_matrix, 0)
         super().__init__(alleles, root_distribution, transition_matrix)
 
 
+class HKY(MutationModel):
+    """
+    The Hasegawa, Kishino and Yano mutation model (Hasegawa et al. 1985).
+
+    .. todo: documentation
+    """
+
+    def __init__(self, ts_tv_ratio=0.5, root_distribution=None):
+        alleles = [b"A", b"C", b"G", b"T"]
+        if root_distribution is None:
+            # ACGT order
+            root_distribution = [0.25, 0.25, 0.25, 0.25]
+
+        transition_matrix = np.full((4, 4), 1.0)
+        np.fill_diagonal(transition_matrix, 0.0)
+        # positions in transition matrix for transversions
+        transition_pos = ((0, 1, 2, 3), (2, 3, 0, 1))
+        kappa = 2 * ts_tv_ratio
+        transition_matrix[transition_pos] = kappa
+        transition_matrix *= root_distribution
+        row_sums = transition_matrix.sum(axis=1, dtype="float64")
+        transition_matrix = transition_matrix / row_sums[:, np.newaxis]
```

Comment:
Ah-ha. This is an infinitesimal rate matrix, not a discrete transition matrix, so normalizing the rows separately changes the model. We are parameterizing things so that the the overall rate matrix, denoted `Q` [here](https://en.wikipedia.org/wiki/Models_of_DNA_evolution#HKY85_model_(Hasegawa,_Kishino_and_Yano_1985)) is defined so that
```
Q[i,j] = mutation_rate * transition_matrix[i,j]
```
for `i != j`. Under the hood, we lay down  mutations at rate `mutation_rate` without looking at the state of each site, then use the corresponding row of the transition_matrix to pick the next state. This means that the `Q[i,i]` entries will correspond to "mutations" that we don't actually include since they don't change the state; so our implementation will be most efficient if these are as small as possible - meaning the smallest one should be zero.  So, let's see, I'd do it like this:
```
         transition_matrix /= max(row_sums)
         row_sums = transition_matrix.sum(axis=1, dtype="float64")
         np.fill_diagonal(transition_matrix, 1.0 - row_sums)
```
... and then we can say in the documentation that the mutation model will be be the HKY rates such that (a) the maximum mutation rate is `mutation_rate` and the stationary distribution is `root_distribution`. (oh wait, I have a thought about the parameterization though; I"ll put it in the discussion).

Reply:
OK, thanks for taking the time to lay this out so clearly. 
Now this part of the [verify_model_ts](https://github.com/tskit-dev/msprime/blob/aa2eca95bfff8a19c41bb0db5e1f077503a158de/tests/test_mutations.py#L915)
`for j, (row, p) in enumerate(zip(transitions, model.transition_matrix)):
            p[j] = 0
            p /= sum(p)
            self.chisquare(row, sum(row) * p)`
makes total sense.